### PR TITLE
GalleryViewer Adjustments

### PR DESCRIPTION
**Pull request by chebureki**
#### Summary
1. Use native video player in favor of custom player
2. Only pan in directions where there is another file (avoids confusion)
3. Make footer a header, so player controls are not overlayed
4. Make header fade and not move

#### Checklist
- [ ] Added or updated unit tests (required for all new features)
- [x] Has UI changes
- [x] Includes text changes and localization file updates (footer became header in localization files, no _actual_ text changes)
- [x] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 
Xiaomi Redmi 9, Android 10
IOS 13 Emulator

#### Release Note
```release-note
Replaced video-player with a native one
```

```release-note
Improved image view panning.
```

### CODE REVIEW

**Comment by enahum** [comment]
Hi @chebureki thanks for this improvements, I'll try and take a look before the EOY, but super quick, can you please revert the changes to the localization files other than english? That is cause of the translation flow, we only modify english in the PR's and the rest is done through the translation server.

Thanks

**Comment by enahum** [comment]
@matthewbirtch from the UI/UX point of view, do we want to move the footer to the top of the screen as in this PR? Should we look for a different alternative?

**Comment by matthewbirtch** [comment]
@enahum from UI/UX point of view, It's too cluttered to have those elements all stacked at the top. It also moves those controls to a harder to reach location. Is there another solution here? I think maybe we need to create a design ticket for this specific issue so we can look at it more closely.

**Comment by enahum** [comment]
@matthewbirtch can you or @Rina-dsg  work with @chebureki to find a good solution for this?

**Comment by chebureki** [comment]
> @matthewbirtch can you or @Rina-dsg work with @chebureki to find a good solution for this?

You can reach me via
email: kirill.sokol@protonmail.com
mattermost-dev: @kirill-sokol

**Comment by Rina-dsg** [comment]
@chebureki I'll reach out to you next week to polish the solution.

**Review by simonojok19 — COMMENTED** [review]
LGTM

**Comment by enahum** [comment]
Hi @Rina-dsg were you able to look at a polished solution for this ?

**Comment by Rina-dsg** [comment]
@enahum yeah, I checked. The bar with an avatar and a share btn should be under the media preview. I messaged all my notes @chebureki directly on MM. Not sure if he updated the pr after.
Screenshot 2021-12-23 at 19 25 28

**Comment by chebureki** [comment]
Sorry, have been busy with holidays and other stuff. I will try to get into it tomorrow :)
Feeling a bit weak after my booster-shot today

**Comment by chebureki** [comment]
bump

File: app/screens/gallery/footer/index.tsx [hunk 1]
```diff
@@ -138,27 +140,18 @@ const Footer = forwardRef<FooterRef, FooterProps>((props: FooterProps, ref) => {
...
-    const translateY = opacity.interpolate({
-        inputRange: [0, 1],
-        outputRange: [99, 0],
-    });
-
-    const downloadingY = downloadingOpacitity.interpolate({
-        inputRange: [0, 1],
-        outputRange: [99, 0],
-    });
-
```
Comment (enahum) [inline app/screens/gallery/footer/index.tsx:150]: why the removal of the animation? I don't think this is expected but let me know your reasoning

File: app/screens/gallery/gallery.js [hunk 2]
```diff
@@ -34,10 +34,10 @@ export default class Gallery extends PureComponent {
 
         this.state = {
             index: props.index,
-            footerVisible: true,
+            headerVisible: true,
         };
 
-        this.footer = React.createRef();
+        this.header = React.createRef();
```
Comment (enahum) [inline app/screens/gallery/gallery.js:40]: this reference is being used in the footer component, I don't think you need to rename it

File: app/screens/gallery/gallery_video.tsx [hunk 3]
```diff
@@ -117,47 +66,24 @@ const GalleryVideo = ({file, deviceHeight, deviceWidth, intl, isActive, showHide
...
-            <VideoControls
-                ref={controlsRef}
-                mainColor={theme?.buttonBg}
-                isLandscape={width > height}
-                paused={paused}
-                onPlayPause={onPlayPause}
-                onSeek={onSeek}
-                showHideHeaderFooter={showHideHeaderFooter}
-            />
+            <View>
```
Comment (enahum) [inline app/screens/gallery/gallery_video.tsx:74]: is this view needed?

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/screens/gallery/gallery_video.tsx (hunk 3 above)
Comment (chebureki) [reply app/screens/gallery/gallery_video.tsx:74]: No, it isn't needed :)

File: app/screens/gallery/footer/index.tsx (hunk 1 above)
Comment (chebureki) [reply app/screens/gallery/footer/index.tsx:150]: Also not needed

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
This is great @chebureki. However, the link and share buttons in the footer don't seem to work unless I tap the very top of them (tested on iOS). I think these buttons might be interfering with the safe area at the bottom and may need to bump up to avoid that interference.

**Comment by chebureki** [comment]
> This is great @chebureki. However, the link and share buttons in the footer don't seem to work unless I tap the very top of them (tested on iOS). I think these buttons might be interfering with the safe area at the bottom and may need to bump up to avoid that interference.

could you by any chance provide a screenshot or recording :). Sorry, I just don't have access to a apple device as of now

**Comment by matthewbirtch** [comment]
> could you by any chance provide a screenshot or recording :). Sorry, I just don't have access to a apple device as of now

Yep, sorry I should have provided that. Here's a screenshot. The red rectangle represents the safe area and as you can see, some of the UI is overlapping in to this area.

image

**Comment by larkox** [comment]
@chebureki thank you very much for your contribution, but we are closing this in favor of the V2 app.

**Pull request closed without merging**